disttask/framework/storage/task_table.go

```diff
@@ -381,11 +381,13 @@ func (stm *TaskManager) GetSubtaskInStatesCnt(taskID int64, states ...interface{
 
 // CollectSubTaskError collects the subtask error.
 func (stm *TaskManager) CollectSubTaskError(taskID int64) ([]error, error) {
-	rs, err := stm.executeSQLWithNewSession(stm.ctx, "select error from mysql.tidb_background_subtask where task_key = %? AND state = %?", taskID, proto.TaskStateFailed)
+	rs, err := stm.executeSQLWithNewSession(
```

Comment:
in which case will subtask be cancelled when the task state is not in(`canceling`, `reverting`)

Reply:
> in which case will subtask be cancelled when the task state is not in(`canceling`, `reverting`)

It is possible when using global sort, the 
```
minimalTasks, err := scheduler.SplitSubtask(ctx, subtask.Meta)
```
May return context.Cancel err.